Tests/SwiftDiagnosticsTest/FixItTests.swift

```diff
@@ -0,0 +1,37 @@
+import SwiftParser
+import SwiftParserDiagnostics
+import SwiftSyntax
+import XCTest
+import _SwiftSyntaxTestSupport
+
+final class FixItTests: XCTestCase {
+  func testEditsForFixIt() {
+    let testCases: [[(range: Range<Int>, replacement: String)]] = [
+        [(9..<15, "Multiident "), (15..<21, "")],
+        [(9..<15, "MultiIdent "), (15..<21, "")]
+      ]
+
+    let markedSource = "protocol Multi 1️⃣ident {}"
+    let (_, source) = extractMarkers(markedSource)
+    let tree = Parser.parse(source: source)
+    let diags = ParseDiagnosticsGenerator.diagnostics(for: tree)
+    XCTAssertEqual(diags.count, 1)
+    let diag = diags.first!
+    XCTAssertEqual(diag.fixIts.count, 2)
+
+    for i in 0..<diag.fixIts.count {
+      let fixIt = diag.fixIts[i]
+      let changes = fixIt.changes
+      let edits = fixIt.edits
+      XCTAssertNotEqual(changes.count, edits.count)
+      let expctedEdits = testCases[i]
+      XCTAssertEqual(expctedEdits.count, edits.count)
+      for j in 0..<edits.count {
+        let edit = edits[j]
+        let expectedEdit = expctedEdits[j]
+        XCTAssertEqual(edit.range, AbsolutePosition(utf8Offset: expectedEdit.range.lowerBound)..<AbsolutePosition(utf8Offset: expectedEdit.range.upperBound))
+        XCTAssertEqual(edit.replacement, expectedEdit.replacement)
+      }
```

Comment:
Shouldn’t you be able to just compare the actual edits with the expected edits using `XCTAssertEqual`?

Reply:
my bad. let me fix it.